flow/probes/ebpf.go

```diff
@@ -73,24 +73,31 @@ func (p *EBPFProbe) run() {
 	_, flowEBPFChan, _ := p.flowTable.Start()
 	defer p.flowTable.Stop()
 
-	ebpfUpdate := time.Duration(p.Ctx.Config.GetInt("agent.flow.ebpf.kernel_scan_interval")) * time.Second
-	flowTableSz := p.Ctx.Config.GetInt("flow.max_entries")
+	ebpfPollingRate := time.Second / time.Duration(p.Ctx.Config.GetInt("agent.flow.ebpf.polling_rate"))
+	const ebpfMaxPollDelay = 10 * time.Second
 
 	var startKTimeNs int64
 	var start time.Time
 
-	updateTicker := time.NewTicker(ebpfUpdate)
-	defer updateTicker.Stop()
+	updateNow := time.NewTicker(ebpfMaxPollDelay)
+	defer updateNow.Stop()
 
 	flowPoolSize := 2 * cap(flowEBPFChan)
 	kernFlows := make([]C.struct_flow, flowPoolSize)
 	ebpfFlows := make([]flow.EBPFFlow, flowPoolSize)
 
 	var key, nextKey C.__u64
 	var nextAvailablePtr int
+	var now time.Time
 	for {
 		select {
-		case now := <-updateTicker.C:
+		case <-p.quit:
+			return
+		case now = <-updateNow.C:
+		default:
+			if now.IsZero() {
```

Comment:
what about now := time.Now() line 91 this will avoid this if each loop

Reply:
yep much better solution